stl/inc/xutility

```diff
@@ -926,6 +933,8 @@ _NODISCARD constexpr decltype(auto) _Get_unwrapped(_Iter&& _It) {
 
 template <class _Iter>
 using _Unwrapped_t = _Remove_cvref_t<decltype(_Get_unwrapped(_STD declval<_Iter>()))>;
+template <class _Iter, bool _Is_unwrapped>
+using _Maybe_unwrapped_t = conditional_t<_Is_unwrapped, _Unwrapped_t<_Iter>, _Remove_cvref_t<_Iter>>;

```

Comment:
Can we reuse `_Maybe_wrapped`?

https://github.com/microsoft/STL/blob/3e046a483ed92f90f7876fc454d672fb2716c45a/stl/inc/ranges#L60-L61